src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingEnchanter.java

```diff
@@ -0,0 +1,360 @@
+    {
+        return ENCHANTER;
+    }
+
+    @Override
+    public String getSchematicName()
+    {
+        return ENCHANTER;
+    }
+
+    @Override
+    public int getMaxBuildingLevel()
+    {
+        return MAX_BUILDING_LEVEL;
+    }
+
+    @Override
+    public BuildingEntry getBuildingRegistryEntry()
+    {
+        return ModBuildings.enchanter;
+    }
+
+    /**
+     * Add a new worker to gather xp from.
+     * @param blockPos the pos of the building.
+     */
+    public void addWorker(final BlockPos blockPos)
+    {
+        buildingToGatherFrom.put(blockPos, false);
+        markDirty();
+    }
+
+    /**
+     * Remove a worker to stop gathering from.
+     * @param blockPos the pos of that worker.
+     */
+    public void removeWorker(final BlockPos blockPos)
+    {
+        buildingToGatherFrom.remove(blockPos);
+        markDirty();
+    }
+
+    /**
+     * Set the daily drainage.
+     * @param qty the quantity to set.
+     */
+    public void setDailyDrainage(final int qty)
+    {
+        this.dailyDrain = qty;
+        markDirty();
+    }
+
+    /**
+     * Get the daily drain
+     * @return the drain
+     */
+    public int getDailyDrain()
+    {
+        return dailyDrain;
+    }
+
+    @Override
+    public void onWakeUp()
+    {
+        final Set<BlockPos> keys = new HashSet<>(buildingToGatherFrom.keySet());
+        buildingToGatherFrom.clear();
+        keys.forEach(k -> buildingToGatherFrom.put(k, false));
+    }
+
+    @Override
+    public void deserializeNBT(final NBTTagCompound compound)
+    {
+        super.deserializeNBT(compound);
+        buildingToGatherFrom.clear();
+        NBTUtils.streamCompound(compound.getTagList(TAG_GATHER_LIST, Constants.NBT.TAG_COMPOUND))
+                                      .map(this::deserializeListElement)
+                                      .forEach(t -> buildingToGatherFrom.put(t.getFirst(), t.getSecond()));
+        dailyDrain = compound.getInteger(TAG_QUANTITY);
+    }
+
+    /**
+     * Helper to deserialize a list element from nbt.
+     * @param nbtTagCompound the compound to deserialize from.
+     * @return the resulting blockPos/boolean tuple.
+     */
+    private Tuple<BlockPos, Boolean> deserializeListElement(final NBTTagCompound nbtTagCompound)
+    {
+        final BlockPos pos = BlockPosUtil.readFromNBT(nbtTagCompound, TAG_POS);
+        final boolean gatheredAlready = nbtTagCompound.getBoolean(TAG_GATHERED_ALREADY);
+        return new Tuple<>(pos, gatheredAlready);
+    }
+
+    @Override
+    public NBTTagCompound serializeNBT()
+    {
+        final NBTTagCompound compound = super.serializeNBT();
+        compound.setTag(TAG_GATHER_LIST, buildingToGatherFrom.entrySet().stream().map(this::serializeListElement).collect(NBTUtils.toNBTTagList()));
+        compound.setInteger(TAG_QUANTITY, dailyDrain);
+        return compound;
+    }
+
+    private NBTTagCompound serializeListElement(final Map.Entry<BlockPos, Boolean> entry)
```

Comment:
Could we possibly abstract or generalise this method out into a utility class? Same with the deserialisation method above.

Reply:
No, it's oddly specific for this case. building a framework for this wouldn't be really beneficial, would only make things difficult to maintain